### PR TITLE
Driver/Minicircuits variable attenuators

### LINKED ISSUE #703: USB variable attenuator - Minicircuits RUDAT-13G-90
For blue force fridges

### PR DESCRIPTION
**Pull request by Dominik-Vogel**
This is the driver for the Minicircuits RCDAT-6000-60 variable attenuator. The PR includes the implementation over Telnet(Ethernet) by @euchas and my additional implementation of a version that communicates via USB.
The USB part is general and is supposed to build the basis for other Minicircuit devices that do not have an ethernet interface such as the  RUDAT-13G-90 variable attenuator (#703)

@jenshnielsen , @WilliamHPNielsen:
 - may I assume that no one is using the telnet version yet and chnage the name of the class?
 - What is your opinion for a location of the USBMixin within the qcodes source tree?

Fixes #703.

Todos
 * implement incremental validator
 * change name, to something more general
 * make example notebook
 * test with RUDAT-13G-90

### CODE REVIEW

**Comment by jenshnielsen** [comment]
Renaming is fine as the other one is not merged yet 
I think leaving the USB mixin where it is is fine. We can move it up when and if other drivers need it

File: qcodes/instrument_drivers/Minicircuits/rcdat_6000_60_usb.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            self.ask_cmd('get_device_model_name'))
+        serial = self._bytearray_to_string(
+            self.ask_cmd('get_device_serial_number'))
+        firmware = self._bytearray_to_string(self.ask_cmd('get_firmware'))
+        return {'vendor': 'Mini-Circuits',
+                'model': model,
+                'serial': serial,
+                'firmware': firmware}
+
+    # this does not work for this instrument for no apparent reason
```
Comment (WilliamHPNielsen) [inline qcodes/instrument_drivers/Minicircuits/rcdat_6000_60_usb.py]: I find this docstring confusing

Comment (Dominik-Vogel) [reply qcodes/instrument_drivers/Minicircuits/rcdat_6000_60_usb.py]: In which way do you think it is confusing? Do you have a suggestion?

Comment (WilliamHPNielsen) [reply qcodes/instrument_drivers/Minicircuits/rcdat_6000_60_usb.py]: Two cases of "this" in a sentence is almost always confusing. And "does not work" can mean a number of things. I am guessing that the following would be more descriptive:

```
# NOTE: the following command silently failed when tested on unit XXXXX
# according to the manual, it should work.
```

Comment (Dominik-Vogel) [reply qcodes/instrument_drivers/Minicircuits/rcdat_6000_60_usb.py]: 👍

**Comment by sohailc** [comment]
Hi @Dominik-Vogel Dominik. My implementation of the driver now also uses pywinusb instead of the microcircuits supplied dll. I think mine is more condensed but perhaps I am missing stuff. Maybe we can discuss this and come up with a 'best of both worlds' implementation? Please have a look at https://github.com/QCoDeS/Qcodes/pull/1073

**Pull request closed without merging**